npm/pkg/controlplane/controllers/v2/podController.go

```diff
@@ -369,6 +377,7 @@ func (c *PodController) syncAddedPod(podObj *corev1.Pod) error {
 	// Create npmPod and add it to the podMap
 	npmPodObj := common.NewNpmPod(podObj)
 	c.podMap[podKey] = npmPodObj
+	metrics.AddPod()
```

Comment:
Goal: call `metrics.AddPod()` once per Pod/IP pair (similar for `metrics.RemovePod()`)


- If a Pod has an update to its IP, follow Pod DELETE then Pod CREATE.
- For Pod DELETE: `c.cleanupDeletePod()` is only called when `podKey` exists in `c.podMap`
- For Pod CREATE: `c.syncAddedPod()` is only called when `podKey` does not exist in `c.podMap`. Otherwise, another codepath is taken.
- Keys are only added/removed from `c.podMap` in `c.syncAddedPod()` and `c.cleanupDeletePod()`